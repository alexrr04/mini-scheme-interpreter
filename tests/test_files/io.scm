(define (even x)
    (= (mod x 2) 0))

(define (collatz x)
    (cond ((= x 1) 1)
          ((even x) (collatz (/ x 2)))
          (#t (collatz (+ (* 3 x) 1)))))

(define (main)
    (display "Enter a number and apply the Collatz Conjecture: ")
    (define x (read))
    (newline)
    (display "You entered: ")
    (display x)
    (newline)
    (display "Applying the Collatz Conjecture returns: ")
    (display (collatz x)))


