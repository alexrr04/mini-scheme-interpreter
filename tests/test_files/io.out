Enter a number and apply the Collatz Conjecture: 
You entered: 2000
Applying the Collatz Conjecture returns: 1